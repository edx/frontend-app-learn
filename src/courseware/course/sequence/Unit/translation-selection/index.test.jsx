import { shallow } from '@edx/react-unit-test-utils';

import TranslationSelection from './index';

jest.mock('@edx/paragon', () => ({
  IconButton: 'IconButton',
  Icon: 'Icon',
  ProductTour: 'ProductTour',
}));
jest.mock('@edx/paragon/icons', () => ({
  Language: 'Language',
}));
jest.mock('./useTranslationTour', () => () => ({
  translationTour: {
    abitrarily: 'defined',
  },
  isOpen: false,
  open: jest.fn().mockName('open'),
  close: jest.fn().mockName('close'),
}));

describe('<TranslationSelection />', () => {
  const props = {
    courseId: 'course-v1:edX+DemoX+Demo_Course',
    selectedLanguage: 'en',
    setSelectedLanguage: jest.fn().mockName('setSelectedLanguage'),
  };
  it('renders', () => {
    const wrapper = shallow(<TranslationSelection {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
