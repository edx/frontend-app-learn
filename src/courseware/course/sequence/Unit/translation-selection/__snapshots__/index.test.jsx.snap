// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TranslationSelection /> renders 1`] = `
<Fragment>
  <ProductTour
    tours={
      Array [
        Object {
          "abitrarily": "defined",
        },
      ]
    }
  />
  <IconButton
    alt="change-language"
    className="mr-2 mb-2 float-right"
    iconAs="Icon"
    id="translation-selection-button"
    onClick={[MockFunction open]}
    src="Language"
    variant="primary"
  />
  <TranslationModal
    close={[MockFunction close]}
    courseId="course-v1:edX+DemoX+Demo_Course"
    isOpen={false}
    selectedLanguage="en"
    setSelectedLanguage={[MockFunction setSelectedLanguage]}
  />
</Fragment>
`;
